src/bls/bls.h

```diff
@@ -215,7 +213,26 @@ class CBLSWrapper
     }
 };
 
-class CBLSId : public CBLSWrapper<uint256, BLS_CURVE_ID_SIZE, CBLSId>
+struct CBLSIdImplicit : public uint256
+{
+    CBLSIdImplicit() {}
+    CBLSIdImplicit(const uint256& id)
+    {
+        memcpy(begin(), id.begin(), sizeof(uint256));
+    }
+    static uint256 FromBytes(const uint8_t* buffer)
+    {
+        uint256 instance;
+        memcpy(instance.begin(), buffer, sizeof(uint256));
+        return instance;
+    }
+    void Serialize(uint8_t* buffer) const
+    {
+        memcpy(buffer, data, sizeof(uint256));
```

Comment:
> 1. Why struct?

Only difference (i know about) is the default visibility for class is `private` whereas its `public` for `struct`. So i picked `struct` in this case because there is no need for `private` and its one line less 😄 Any specific concerns/points against `struct`?

> 2. Should use proper types

Makes sense, changed in 1a55d8a

Reply:
>Any specific concerns/points against struct?

That's just me being old-school I guess and expecting `struct`s to be function-less PODs 😄  But on a second thought, it's just ctors and (de)serialization methods, so it kind of fits too. ok then 🙂